### PR TITLE
#184581338 Admin should be able to disable an account for various reasons

### PR DESCRIPTION
**Pull request by Martha-Iradukunda**
#### What does this PR do?
* The PR would likely be adding the feature for an admin to disable a user account and send an email to the user with reasons for the block.
#### Description of Task to be completed?
* The task is to add a feature that allows an admin to disable a user account for various reasons such as fraud, abuse, malicious use or violation of terms of service. The admin should be able to trigger an action to block usage of a particular account and an email should be sent to the user with reasons for the block. Sample violation reasons are provided such as merchants selling illegal items, harassment of other users of the application, and requests from law enforcement.
#### How should this be manually tested?
* Create an admin account and multiple user accounts on the system.
* As the admin, log in to the system and navigate to the user account that needs to be disabled.
* Trigger the action to block usage of the user account.
* Check that the user account has been disabled and the user has received an email with the reasons for the block.
*  Verify that the user is unable to log in to the system.
* PATCH/users/disable/{userId}
#### Any background context you want to provide?
* N/A
#### What are the relevant pivotal tracker/Trello stories?
* [#184581338](https://www.pivotaltracker.com/n/projects/2632830/stories/184581338)

#### Screenshots (if appropriate)
* 
Screenshot from 2023-03-20 23-12-17
* 
Screenshot from 2023-03-21 22-09-42

#### Questions:
N/A

### CODE REVIEW

**Review by edwinnambaje — CHANGES_REQUESTED** [review]
You can first solve the tests issues and after you can clearly show us the working in the documentation

File: src/controllers/users.controllers.js [hunk 1]
```diff
@@ -381,4 +384,38 @@ export default class Users {
...
+        content: `Dear ${user.username},<br/>
+        Your account has been disabled due to the following reason(s):<br/>
+        ${user.disabledReason}<br/>Please contact the support team if you have any questions.<br/>Best regards, <br/>The support team`,
+      }));
+
+      res.status(200).json({ message: 'User account disabled successfully' });
+
+      // res.json(user);
+    } catch (error) {
+      console.error(error);
```
Comment (nshutijonathan) [inline src/controllers/users.controllers.js]: remove `console.log`

**Review by nshutijonathan — COMMENTED** [review]
remove `console.log`

**Review by blackd44 — APPROVED** [review]
LGTM

**Review by edwinnambaje — APPROVED** [review]
LGTM

**Review by bencyubahiro77 — APPROVED** [review]

**Review by Patienceineza — APPROVED** [review]
LGTM

**Review by Zeden12 — APPROVED** [review]
LGTM

File: src/controllers/users.controllers.js [hunk 2]
```diff
@@ -399,4 +402,38 @@ export default class Users {
...
```
Comment (nshutijonathan) [inline src/controllers/users.controllers.js]: remove `console.log`

**Review by nshutijonathan — COMMENTED** [review]
remove console.log

File: src/controllers/users.controllers.js [hunk 3]
```diff
@@ -399,4 +402,37 @@ export default class Users {
...
+      user.disabledUser = true;
+      user.disabledReason = req.body.reason;
+
+      await user.save();
+
+      sendEmail(emailConfig({
+        email: user.email,
+        subject: 'Your account has been disabled',
+        // eslint-disable-next-line no-undef
+        content: `Dear ${user.username},<br/>
```
Comment (UmuhireJessie) [inline src/controllers/users.controllers.js]: Good job here @Martha-Iradukunda!
For consistency, make sure the email sent to the users looks the same.

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
See my comment

**Comment by Martha-Iradukunda** [comment]
I don't understand?

**Review by UmuhireJessie — APPROVED** [review]

**Pull request merged**